feat(kyc): enhance KycBlock with risk labels and KycDecision component

tomer-shvadron opened the pull request:
- Add risk labels extraction for better KYC insights
- Introduce new KycDecision component to display KYC results and issues
- Refactor existing logic to streamline data processing and state representation


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enhanced KYC risk display with clear visual badges indicating risk levels, including a default state when no risks are detected.
  
- **Refactor**
  - Streamlined decision mapping and error reporting for improved service stability.
  - Optimized workflow processing by simplifying method calls and removing obsolete steps.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

## Screenshots

![image](https://github.com/user-attachments/assets/975e7c48-3fbe-4f2e-9968-56407832df0d)
![image](https://github.com/user-attachments/assets/eee2c216-da77-466b-b2b0-993880d663c2)

tomer-shvadron committed:
feat(kyc): enhance KycBlock with risk labels and KycDecision component

- Add risk labels extraction for better KYC insights
- Introduce new KycDecision component to display KYC results and issues
- Refactor existing logic to streamline data processing and state representation

Omri-Levy commented on apps/backoffice-v2/src/lib/blocks/components/KycBlock/hooks/useKycBlock/useKycBlock.tsx:
```diff
@@ -55,46 +55,66 @@ export const useKycBlock = ({
   }, [allDocuments]);
 
+  const riskLabels = kycSessionKeys?.length
+    ? kycSessionKeys.flatMap(key =>
+        key === 'invokedAt'
+          ? []
+          : childWorkflow?.context?.pluginsOutput?.kyc_session[key]?.result?.decision?.riskLabels
+              ?.length
+          ? childWorkflow?.context?.pluginsOutput?.kyc_session[key]?.result?.decision?.riskLabels
+          : 'none',
+      )
```
Can we turn this into if conditions? This is a bit hard to read. 🙏🏼

tomer-shvadron replied:
WOW, totally justified. That's what happens when you're working on auto-pilot at 11pm! Thanks dude

Omri-Levy commented on apps/backoffice-v2/src/lib/blocks/components/KycBlock/hooks/useKycBlock/useKycBlock.tsx:
```diff
@@ -254,8 +272,7 @@ export const useKycBlock = ({
         })
-        .build()
-        .flat(1);
+        .buildFlat();
```
Appreciated! ❤️

Omri-Levy commented on apps/backoffice-v2/src/lib/blocks/components/KycDecision/KycDecision.tsx:
```diff
@@ -0,0 +1,86 @@
+  documentImageQualitySufficient: 'Document image quality insufficient',
+} as const;
+
+export const KycDecision: FunctionComponent<ExtractCellProps<'kycDecision'>> = ({
```
I don't understand why a cell was created for this. Cells are supposed to be primitives used for composition and re-use. Is there a plan to use it in multiple blocks or multiple times within a block etc?

tomer-shvadron replied:
The issue I faced was showing the "issues" as a list of badges. I found it too hard and confusing to add that functionality in `EditableDetails` that's already pretty robust. I think that creating the new cell that renders `EditableDetails` and the needed badges list the easiest, quickest and most clear way to achieve readable and maintainable code.

tomer-shvadron committed:
fix(workflow): correct spread operator in ALL_KNOWN_CHECKS constant

- Update ALL_KNOWN_CHECKS to properly include DECISION_CHECKS
- Ensure correct array structure for decision checks integration

tomer-shvadron committed:
feat: pR comment fix

tomer-shvadron committed:
feat(kyc): refactor KYC block to display risk labels using badges

- Add badge component to show risk labels with descriptive statuses
- Remove the KycDecision component as it is no longer needed
- Update type definitions and adjust data handling in useKycBlock
